### ISSUE TITLE
Apollo Federation types missing on generate

### ISSUE THREAD

**Issue by mshamasa** [dotansimha/graphql-code-generator#7513]
**Describe the bug**
I am using the latest Apollo Federation and when I generate all the types based on my schema, it looks good and types everything. But I noticed when looking at the generated file, a couple of Federation types are missing.

type __Service
type Query_EntitiesArgs

I have moved

`federation: true` below `useIndexSignature` and ran generate and still have the same issue. If I remove the federation config flag, it will generate the above 2 types.

This doesn't cause any issues on the project or running in dev since I ignore the file, but it seems like odd behavior.

**To Reproduce**
- Repo: https://github.com/mshamasa/federation
- remove `federation` flag
- npm run generate:types
- see type _Service exists
- add `federation` flag
- npm run generate:types
- see type __Service does not generate

1. My GraphQL schema:

```graphql
import { gql } from 'apollo-server';

const typeDefs = gql`
  extend type Query {
    User(ID: Int): User
  }
  type User @key(fields: "ID") {
    ID: Int!
  }
`;

export default typeDefs;
```

2. My GraphQL operations:

```graphql
import type { Resolvers } from '../../../../../../types';

const resolvers: Resolvers = {
  Query: {
    User: async (_root, args) => {
      const { ID } = args;

      return {
        ID,
        name: 'Query#User#hello world!',
      };
    },
  },
  User: {
    __resolveReference: (User) => {
      return {
        ID: User.ID,
        name: '__resolveReference#hello world!',
      };
    },
  },
};

export default resolvers;
```

2. My `codegen.yml` config file:

```yml
overwrite: true
schema: './services/users/src/schemas/index.ts'
config:
  federation: true
generates:
  ./types/graphql/generated.ts:
    plugins:
      - typescript
      - typescript-resolvers
    config:
      useTypeImports: true
      useIndexSignature: true
require:
  - 'ts-node/register/transpile-only'

```

**Expected behavior**
To generate 
- type __Service
- type Query_EntitiesArgs

**Environment:**

- OS: Mac
- `@graphql-codegen/...`: 
  - "@graphql-codegen/cli": "2.5.0",
  - "@graphql-codegen/typescript": "2.4.3",
  - "@graphql-codegen/typescript-resolvers": "2.5.0",
- NodeJS: v16.10.0

**Additional context**
<img width="505" alt="Screen Shot 2022-02-10 at 12 28 37 PM" src="https://user-images.githubusercontent.com/6489276/153491204-47c5c276-8107-4613-88a4-9d71cdde125d.png">
<img width="716" alt="Screen Shot 2022-02-10 at 12 28 50 PM" src="https://user-images.githubusercontent.com/6489276/153491212-30d32e76-7aa3-46be-b608-3506387a9676.png">

**Comment by charlypoly** [comment]
Hi @mshamasa, and thank you for this report! 

Sorry but I'm not adding a lot here but just labeling it according to our new Contribution Guide and [issue flow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

Thanks to your reproduction, it seems you already got into `stage 1` thanks to your reproduction! Thank you for that!

Now to advance to `stage 2` we'll need a failing test; it would be great if someone could help progress the issues through the [stages](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

Thank you, and sorry that this comment is not a complete solution (yet).

**Comment by Tsymalyi** [comment]
Have some prblem with Apollo Federation 2

**Comment by mshamasa** [comment]
I found the issue, it was in my export default, I should have been exporting `typeDefs` as the default. Once I made that change it worked perfectly fine, thank you!

It could be a good idea to throw an error if the `resolvers` are added to the codegen? Thank you!

<img width="1623" alt="Screen Shot 2022-07-19 at 8 20 36 PM" src="https://user-images.githubusercontent.com/6489276/179889455-9259f75b-7ca3-4dbb-8d3b-41121b1658ab.png">
<img width="964" alt="Screen Shot 2022-07-19 at 8 20 27 PM" src="https://user-images.githubusercontent.com/6489276/179889463-3937f40a-4a15-4c2a-8a34-44ecaa716354.png">
